utilities/helpers/postHelper.js

```diff
@@ -195,6 +196,38 @@ const fillReblogs = async (posts = [], userName) => {
     }
   }
 };
+const jsonParse = (post) => {
+  try {
+    return JSON.parse(post.json_metadata);
+  } catch (error) {
+    return null;
+  }
+};
+
+const additionalSponsorObligations = async (posts) => {
+  if (_.isEmpty(posts)) return;
+  for (const post of posts) {
+    const metadata = post.json_metadata ? jsonParse(post) : null;
+    const _id = _.get(metadata, 'campaignId');
+    // if post metadata doesn't have campaignId it's not review
+    if (!_id) continue;
+
+    const { result: campaign } = await Campaign.findOne({ _id });
+    const { result: upvote } = await botUpvoteModel
+      .findOne({ author: post.author, permlink: post.permlink });
+    if (!campaign || !upvote) continue;
+
+    const user = _.find(campaign.users, (u) => u.permlink === upvote.reservationPermlink);
+    const { result: histories } = await paymentHistory.findByCondition({ 'details.reservation_permlink': _.get(user, 'permlink') });
+    if (!user || !histories.length) continue;
+
+    const rewards = _.filter(histories, (history) => _.includes(['beneficiary_fee', 'review'], history.type));
+    const rewardSponsor = _.sumBy(rewards, 'amount')
+      * _.get(user, 'hiveCurrency', _.get(rewards, '[0].details.hiveCurrency'));
```

Comment:
why all this if you can just take the field totalVotesAmount and hive currency from reservation, you don't need to take all the rewards, you only need the sum of votes